src/scripts/store.ts

```diff
@@ -0,0 +1,83 @@
+import User, { isValidUser } from './user.ts';
+
+const USERS_URL = 'https://dummyjson.com/users';
+
+export default class Store {
+    #users: Array<User>;
+    #localUsersKey: string;
+    #renderUsers: (users: Array<User>) => void;
+
+    constructor(
+        localUsersKey: string,
+        renderUsers: (users: Array<User>) => void
+    ) {
+        this.#localUsersKey = localUsersKey;
+        this.#renderUsers = renderUsers;
+
+        this.#users = this.getLocalUsers();
+        this.#renderUsers(this.#users);
+
+        this.fetchUsers().then((users) => {
+            this.#users.push(...users);
+            this.#renderUsers(this.#users);
+        });
+    }
+
+    // Extract valid user from unknown Array.
+    extractValidUser(usersUnknown: unknown): Array<User> {
+        const users: Array<User> = [];
+        if (Array.isArray(usersUnknown)) {
+            usersUnknown.forEach((user: unknown) => {
+                if (isValidUser(user)) {
+                    users.push(
+                        new User(
+                            user.id,
+                            user.firstName,
+                            user.lastName,
+                            user.age,
+                            user.email,
+                            user.phone,
+                            user.gender
+                        )
+                    );
+                }
+            });
+        }
+
+        return users;
+
+    }
+
+    // Fetch locally stored users.
+    getLocalUsers(): Array<User> {
+        let localUsers: unknown;
+        try {
+            const localUsersStr = localStorage.getItem(this.#localUsersKey);
+            if (localUsersStr) {
+                localUsers = JSON.parse(localUsersStr);
+            }
+        } catch (err) {
+            console.log(err);
+        }
+
+        return this.extractValidUser(localUsers);
+    }
```

Comment:
Can we re-write this logic something like this?

```suggestion
    getLocalUsers(): Array<User>{
    try {
        const localUsersStr = localStorage.getItem(this.#localUsersKey);
        const localUsers = localUsersStr ? JSON.parse(localUsersStr) : [];
        return this.extractValidUser(localUsers);
    } catch (err) {
        // we can use the console.error as well
        console.error("Error parsing local users:", err);
        return [];
    }
}
```

Reply:
Updated it.
https://github.com/dhruvdhaduk-simform/user-management-ts/pull/4/commits/8129c2100e060442e076c9d7d0569e7a2dfa493b